news-catalyst/google-app-scripts
# Hasura get article

Closes #197 

This PR replaces the function "getArticleMeta" and all related code (hopefully - I'll do a bigger sweep of the code when I've gotten all the functionality done) with a version backed by Hasura:

1. opening the "Publishing Tools" sidebar sends a request to look up article data based on the google doc ID and locale
2. this uses the new tables I added this morning that link an article with a google document in Hasura 
3. the following article data is requested: translations of content, slug, tags, authors and category
4. additional data: locales for the org, all categories, all tags, all authors (to populate various select menus)
5. the data is filled into the sidebar form elements where appropriate

Note: this is the first of many PRs for porting the sidebar publishing tools to Hasura. This is complicated so I'm trying to do it step by step. I'm also deleting as much code as possible in an effort to clean things up and keep it all straightforward. You can't currently do anything other than open the sidebar, see the data, and select a tag at the moment. Publishing buttons, preview, translate tools all coming soon.

To test:

because I haven't written the code to save into Hasura yet, you have to use this particular google doc that I've setup data for: "Here’s how Philadelphia public schools will operate this fall"

* Open [the script editor](https://script.google.com/a/newscatalyst.org/d/1ILURq69o3cYUy6k1n1X6HwxdMfl9xWNhILYuZxgLfeblb3IR15WCMZSj/edit)
* Run > Test as add-on... -> select the one option and [Test]
* Open the Publishing Tools Sidebar
* Hopefully, see the data (currently authors are blank; tags aren't assigned; custom byline is blank; the rest are set)

<img width="321" alt="Screen Shot 2021-02-15 at 10 47 27 am" src="https://user-images.githubusercontent.com/3397/107892739-3c675e80-6f7b-11eb-8289-a7c1c72e203a.png">
